chemprop/uncertainty/uncertainty_evaluator.py

```diff
@@ -398,6 +398,142 @@ def evaluate(
         return spearman_coeffs
 
 
+class ConformalRegressionEvaluator(UncertaintyEvaluator):
+    """
+    A class for evaluating the coverage of conformal regression intervals.
+    """
+
+    def raise_argument_errors(self):
+        super().raise_argument_errors()
+        if self.dataset_type != "regression":
+            raise ValueError(
+                "Conformal Regression Evaluator is only for regression dataset types."
+            )
+
+    def evaluate(
+        self,
+        targets: List[List[float]], # shape (data, tasks)
+        preds: List[List[float]],
+        uncertainties: List[List[float]], # shape (data, 2*tasks)
+        mask: List[List[bool]],
+    ):
+        """
+        Args:
+            targets: shape(data, tasks)
+            preds: shape(data, tasks, num_classes)
+            uncertainties: shape(data, 2*tasks, num_classes)
+            mask: shape(data, tasks)
+
+        Returns:
+            Conformal coverage for each task
+        """
+        targets = np.array(targets)
+        mask = np.array(mask, dtype=bool)
+        uncertainties = np.array(uncertainties)
+        num_tasks = uncertainties.shape[1]//2
+        results = []
+
+        for task_id in range(num_tasks):
+            unc_task_id_lower = uncertainties[mask[:, task_id], task_id]
+            unc_task_id_upper = uncertainties[mask[:, task_id], task_id + num_tasks]
+            targets_task_id = targets[mask[:, task_id], task_id]
+            task_results = np.logical_and(unc_task_id_lower <= targets_task_id, targets_task_id <= unc_task_id_upper)
+            results.append(task_results.sum() / task_results.shape[0])
+
+        return results
```

Comment:
```suggestion
        num_tasks = uncertainties.shape[1] // 2
        coverages = []

        for i in range(num_tasks):
            task_unc_lower = uncertainties[mask[:, i], i]
            task_unc_upper = uncertainties[mask[:, i], i + num_tasks]
            task_targets = targets[mask[:, i], i]
            task_results = np.logical_and(task_unc_lower <= task_targets, task_targets <= task_unc_upper)
            coverage = task_results.sum() / task_results.shape[0]
            coverages.append(coverage)

        return coverages
```

Reply:
I changed the format for the other conformal coverage functions to match this style